pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -54,11 +54,14 @@ func (d *Deployment) Deadline() time.Duration {
 
 func (d *Deployment) UpdateStatus(details string, err error) {
 	updated := newStatus(details, err)
-	if !d.status.Equal(updated) {
-		d.status = updated
-		if strings.Contains(details, rollOutSuccess) || isErrAndNotRetryAble(err) {
-			d.done = true
-		}
+	if d.status.Equal(updated) {
+		d.status.changed = false
```

Comment:
Is it possible to have several status updates before the status has been reported in `ReportSinceLastUpdated()`?  Consider the following:
```
d.UpdateStatus("status1")
//  status.changed now true
d.UpdateStatus("status2")
// status.changed now true
d.UpdateStatus("status2")
// status.changed now false
```

If something checks `ReportSinceLastUpdated()` only now, then nothing will be output.

Reply:
Currently, we call `ReportSinceLastUpdated` every 500 ms and deployment statuses are polled every 100ms. So it is possible to have 3-4 updates in that time.

Reply:
i will add this change back in.